content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Virtual IPs and Service Proxies
+content_type: reference
+---
+
+<!-- overview -->
+Every {{< glossary_tooltip term_id="node" text="node" >}} in a Kubernetes
+cluster runs a [kube-proxy](/docs/reference/command-line-tools-reference/kube-proxy/)
+(unless you have deployed your own alternative component in place of `kube-proxy`).
+
+The `kube-proxy` component is responsible for implementing a _virtual IP_
+mechanism for {{< glossary_tooltip term_id="service" text="Services">}}
+of `type` other than
+[`ExternalName`](/docs/concepts/services-networking/service/#externalname).
+
+<!-- body -->
+
+## Motivation
+
+A question that pops up every now and then is why Kubernetes relies on
+proxying to forward inbound traffic to backends. What about other
+approaches? For example, would it be possible to configure DNS records that
+have multiple A values (or AAAA for IPv6), and rely on round-robin name
+resolution?
+
+There are a few reasons for using proxying for Services:
+
+* There is a long history of DNS implementations not respecting record TTLs,
+  and caching the results of name lookups after they should have expired.
+* Some apps do DNS lookups only once and cache the results indefinitely.
+* Even if apps and libraries did proper re-resolution, the low or zero TTLs
+  on the DNS records could impose a high load on DNS that then becomes
+  difficult to manage.
+
+Later in this page you can read about various kube-proxy implementations work. Overall,
+you should note that, when running `kube-proxy`, kernel level rules may be modified
+(for example, iptables rules might get created), which won't get cleaned up, in some
+cases until you reboot.  Thus, running kube-proxy is something that should only be done
+by an administrator which understands the consequences of having a low level, privileged
+network proxying service on a computer.  Although the `kube-proxy` executable supports a
+`cleanup` function, this function is not an official feature and thus is only available
+to use as-is.
+
+### Configuration
+
+Note that the kube-proxy starts up in different modes, which are determined by its configuration.
+
+- The kube-proxy's configuration is done via a ConfigMap, and the ConfigMap for
+  kube-proxy effectively deprecates the behaviour for almost all of the flags for
+  the kube-proxy.
+- The ConfigMap for the kube-proxy does not support live reloading of configuration.
+- The ConfigMap parameters for the kube-proxy cannot all be validated and verified
+  on startup.  For example, if your operating system doesn't allow you to run iptables
+  commands, the standard kernel kube-proxy implementation will not work.  Likewise,
+  if you have an operating system which doesn't support `netsh`, it will not run in
+  Windows userspace mode.
```

Comment:
configmap is the most common configuration more, but not the only on

Reply:
This is a refactor PR. If the facts are wrong, we should fix the existing page - https://kubernetes.io/docs/concepts/services-networking/service/#configuration - and I can rebase.

The bugfix should not come via a refactor PR; partly that's my view on what refactoring _is_, and there's another reason: refactor PRs of this size can take between one and nine months to merge, which is a long time to wait for a docs bugfix.

(although, to be fair, I suspect that the existing page has been wrong for some years).

@aojea would you be willing to file an issue about the inaccuracy?

Reply:
hmm, due to the history we have with flags vs config in kube-proxy (flags doesn't work fine) I think that is better to keep it this way unless we have a better alternative ... you can use other configuration modes, but having consistency will make a better user experience

Reply:
I'll mark this as resolved as we're doing feedback on the broken-out PRs now.